code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -50,15 +54,19 @@
 	if(!.)
 		return FALSE
 
-	alerted = null
+	clear_alerted()
 	var/do_alert = (COOLDOWN_FINISHED(src, alert_cooldown) && (locate(/mob/living) in contents))
 	if(!do_alert)
-
 		return TRUE
+
 	// Cache the list before we open the box.
-	alerted = viewers(7, src)
-	// There are no mobs to alert? clear the list & prevent further action after opening the box
-	if(!(locate(/mob/living) in alerted))
+	alerted = list()
+	for(var/mob/living/alerted_mob in viewers(7, src))
+		alerted += alerted_mob
+		RegisterSignal(alerted_mob, COMSIG_QDELETING, PROC_REF(remove_from_alerted)) // so we don't get hard dels
```

Comment:
There is no need to register signals and stuff. Inside the after_open() proc once we have finished iterating over the list immediately clear it I.e. alerted.Cut(). We don't need this list after the after_open() proc is done with it

Reply:
This also means your entire PR can be just that one line. No need to use Destroy() or create an clear_alerted() proc or any of that.

I remember refactoring closets a while back so it should work

Reply:
https://github.com/tgstation/tgstation/blob/877b1ce0d24dea6472a5e4f7fb9e7201cc561e60/code/game/objects/structures/crates_lockers/closets/cardboardbox.dm#L63

Replace with alerted.Cut() we don't want to allocate a new list every time. This was my bad

Reply:
> Good point, I'm not really sure why it was being done this way in the first place when they were just clearing it right away anyway.

I was a noob coder back then. I still am 👍